tests/5.1/teams/test_target_set_num_teams.c

```diff
@@ -0,0 +1,45 @@
+//===--------------------- test_target_set_num_teams.c ----------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test uses the omp_set_num_teams utility to set the default number of
+// teams allocated for team clauses. The test runs teams and checks to see if
+// the number allocated is correct and reflective of the new default set. It
+// then checks to see if an explicitly stated num_teams clause overrides
+// the omp_set_num_teams properly.
+//
+//===------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+int main() {
+
+
+	int errors = 0; 
+	int num_teams = 0;
+	omp_set_num_teams(4);
+        	
+	#pragma omp teams 
+	{
+		if (omp_get_team_num() == 0 ) {
+			num_teams = omp_get_num_teams();
+		}
+	}               
+	
+	OMPVV_ERROR_IF(num_teams != 4, "Number of teams detected was not the number set by omp_set_num_teams()");
+	OMPVV_TEST_AND_SET(errors, num_teams != 4);
+
+	#pragma omp teams num_teams(OMPVV_NUM_TEAMS_HOST)
+	{
+		if (omp_get_team_num() == 0 ) {
+			num_teams = omp_get_num_teams();
+		}
+	}
+
+	OMPVV_ERROR_IF(num_teams != OMPVV_NUM_TEAMS_HOST, "Number of teams was not properly overriden by the num_teams clause");	
```

Comment:
```suggestion
	OMPVV_ERROR_IF(num_teams <= OMPVV_NUM_TEAMS_HOST, "Number of teams was not properly overriden by the num_teams clause");	
```
pg. 101 5.1 spec
16 If the num_teams clause is present, lower-bound is the specified lower bound and upper-bound is
17 the specified upper bound on the number of teams requested. If a lower bound is not specified, the
18 lower bound is set to the specified upper bound. The number of teams created is implementation
19 defined, but it will be greater than or equal to the lower bound and less than or equal to the upper
20 bound.
Because only an upperbound is present, the number of teams created is allowable to be less than or equal to the upperbound.